pkg/snapshotmgr/snapshot_manager.go

```diff
@@ -526,7 +526,7 @@ func (this *SnapshotManager) isTerminalState(uploadCR *v1api.Upload) bool {
 	if uploadCR == nil {
 		return true
 	}
-	return uploadCR.Status.Phase == v1api.UploadPhaseCompleted || uploadCR.Status.Phase == v1api.UploadPhaseCleanupFailed || uploadCR.Status.Phase == v1api.UploadPhaseCanceled
+	return uploadCR.Status.Phase == v1api.UploadPhaseCompleted || uploadCR.Status.Phase == v1api.UploadPhaseCanceled
```

Comment:
UploadPhaseCleanupFailed is actually a terminal state wrt to the actual upload, leave it as is.